destination/config.go

```diff
@@ -0,0 +1,93 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package destination
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/conduitio-labs/conduit-connector-nats-jetstream/config"
+	"github.com/conduitio-labs/conduit-connector-nats-jetstream/validator"
+)
+
+const (
+	// defaultRetryWait is the default retry wait time when ErrNoResponders is encountered.
+	defaultRetryWait = time.Second * 5
+	// defaultRetryAttempts is the retry number of attempts when ErrNoResponders is encountered.
+	defaultRetryAttempts = 3
+)
+
+const (
+	// ConfigKeyRetryWait is a config name for a retry wait duration.
+	ConfigKeyRetryWait = "retryWait"
+	// ConfigKeyRetryAttempts is a config name for a retry attempts count.
+	ConfigKeyRetryAttempts = "retryAttempts"
+)
+
+// Config holds destination specific configurable values.
+type Config struct {
+	config.Config
+
+	RetryWait     time.Duration `key:"retryWait"`
+	RetryAttempts int           `key:"retryAttempts"`
+}
+
+// Parse maps the incoming map to the Config and validates it.
+func Parse(cfg map[string]string) (Config, error) {
```

Comment:
WDYT about using JSON unmarshaling to try simplifying the parsing logic?

Reply:
I was thinking of marshalling the map into JSON bytes, and then unmarshalling into a struct. It's an additional step, but  I think it would save all the lines where parse fields one by one (parse int, parse duration etc.)

Reply:
But if we have, for example:
```go
cfg := map[string]string{
    "retryWait":     "1000",
    "retryAttempts": "10",
}
```

After marshalling the map into JSON bytes, we will have the following data (converted to a string): `{"retryWait": "1000", "retryAttempts": "10"}`. 
But unmarshalling it to a struct will fail trying to unmarshal string into integer, or string into `time.Duration`, etc.

Reply:
Ah, good point. :+1: